### ISSUE TITLE
Spurious SC2086 warning on $PIPESTATUS

### ISSUE THREAD

**Issue by viavi-ab** [koalaman/shellcheck#3145]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/bash
# shellcheck disable=SC2181 # (style): Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

[ $? -eq 0 ]
[ ${PIPESTATUS[0]} -eq 0 ]
```

#### Here's what shellcheck currently says:
```
Line 5	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

Nothing. Elements of the $PIPESTATUS array are numeric by definition. Just like $?, it doesn't need to doublequoted (at least not to prevent globbing and word splitting; you may want to quote it in case the particular element you're accessing is empty)

**Comment by wileyhy** [comment]
When people often fail to remember a rule, it's less useful to teach a
rule's exceptions, in my opinion.

Single bracket test operates based on the number of opt-args, so using
unquoted variables in this context can alter how `[` differentiates
operators from operands.
